warehouse/jobs/runner.go

```diff
@@ -0,0 +1,373 @@
+					}
+					if resp.Status == pgnotifier.AbortedState {
+						pkgLogger.Errorf("[WH]: Error in running async task %v", resp.Error)
+						sampleError := fmt.Errorf(resp.Error)
+						asyncWhJob.updateSingleAsyncJob(&asyncjobpayloads[ind], WhJobAborted, sampleError.Error())
+						continue
+					}
+					if resp.Status == pgnotifier.FailedState {
+						pkgLogger.Errorf("[WH]: Error in running async task %v", resp.Error)
+						sampleError := fmt.Errorf(resp.Error)
+						asyncWhJob.updateSingleAsyncJob(&asyncjobpayloads[ind], WhJobFailed, sampleError.Error())
+						continue
+					}
+					successfulAsyncJobs = append(successfulAsyncJobs, asyncjobpayloads[ind])
+
+				}
+				asyncWhJob.updateMultipleAsyncJobs(&successfulAsyncJobs, WhJobSucceeded, "")
+				wg.Done()
+			}()
+			wg.Wait()
+		}
+	}
+}
+
+//Queries the jobsDB and gets active async job and returns it in a
+func (asyncWhJob *AsyncJobWhT) getPendingAsyncJobs(ctx context.Context) ([]AsyncJobPayloadT, error) {
+	var asyncjobpayloads = make([]AsyncJobPayloadT, 0)
+	//Filter to get most recent row for the sourceId/destinationID combo and remaining ones should relegated to aborted.
+	query := fmt.Sprintf(
+		`select 
+	id,
+	sourceid,
+	destinationid,
+	status,
+	jobrunid,
+	COALESCE(starttime, ''),
+	created_at,
+	updated_at,
+	destination_type,
+	taskrunid,
+	tablename,
+	jobtype,
+	async_job_type,
+	namespace from %s where status='%s' LIMIT %d`, warehouseutils.WarehouseAsyncJobTable, WhJobWaiting, MaxBatchSizeToProcess)
+	rows, err := asyncWhJob.dbHandle.Query(query)
+	if err != nil {
+		return asyncjobpayloads, err
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var asyncjobpayload AsyncJobPayloadT
+		var status string
+		var updated_at, created_at string
+		err = rows.Scan(
+			&asyncjobpayload.Id,
+			&asyncjobpayload.SourceID,
+			&asyncjobpayload.DestinationID,
+			&status,
+			&asyncjobpayload.JobRunID,
+			&asyncjobpayload.StartTime,
+			&created_at,
+			&updated_at,
+			&asyncjobpayload.DestType,
+			&asyncjobpayload.TaskRunID,
+			&asyncjobpayload.TableName,
+			&asyncjobpayload.JobType,
+			&asyncjobpayload.AsyncJobType,
+			&asyncjobpayload.Namespace,
+		)
+		if err != nil {
+			asyncWhJob.log.Errorf("WH-Jobs: Error scanning rows %s\n", err)
+			return asyncjobpayloads, err
+		}
+		asyncjobpayloads = append(asyncjobpayloads, asyncjobpayload)
+	}
+	return asyncjobpayloads, nil
+}
+
+//Updates the warehouse async jobs with the status sent as a parameter
+func (asyncWhJob *AsyncJobWhT) updateMultipleAsyncJobs(payloads *[]AsyncJobPayloadT, status string, errMessage string) {
+	asyncWhJob.log.Info("WH-Jobs: Updating pending wh async jobs to Executing")
+	for _, payload := range *payloads {
+		asyncWhJob.updateSingleAsyncJob(&payload, status, errMessage)
+	}
+}
+
+//The above function and the following function can be merged into one. This can be part of future work
+func (asyncWhJob *AsyncJobWhT) updateSingleAsyncJob(payload *AsyncJobPayloadT, status string, errMessage string) {
```

Comment:
Can we rename the function as updateAsyncJob since it's clear it will handle one single Job.

Reply:
Done